models/astrocyte_lr_1994.cpp

```diff
@@ -0,0 +1,543 @@
+  insert_( names::Ca, &astrocyte_lr_1994::get_y_elem_< astrocyte_lr_1994::State_::Ca > );
+  insert_( names::h_IP3R, &astrocyte_lr_1994::get_y_elem_< astrocyte_lr_1994::State_::h_IP3R > );
+}
+
+extern "C" int
+astrocyte_lr_1994_dynamics( double time, const double y[], double f[], void* pnode )
+{
+  // a shorthand
+  typedef nest::astrocyte_lr_1994::State_ S;
+
+  // get access to node so we can almost work as in a member function
+  assert( pnode );
+  const nest::astrocyte_lr_1994& node = *( reinterpret_cast< nest::astrocyte_lr_1994* >( pnode ) );
+
+  // y[] here is---and must be---the state vector supplied by the integrator,
+  // not the state vector in the node, node.S_.y[].
+
+  // The following code is verbose for the sake of clarity. We assume that a
+  // good compiler will optimize the verbosity away ...
+
+  // shorthand for state variables
+  const double& ip3 = y[ S::IP3 ];
+  const double& calc = y[ S::Ca ];
+  const double& h_ip3r = y[ S::h_IP3R ];
+
+  const double alpha_h_ip3r =
+    node.P_.k_IP3R_ * node.P_.Kd_inh_ * ( ip3 + node.P_.Kd_IP3_1_ ) / ( ip3 + node.P_.Kd_IP3_2_ );
+  const double beta_h_ip3r = node.P_.k_IP3R_ * calc;
+  const double J_pump =
+    node.P_.rate_SERCA_ * std::pow( calc, 2 ) / ( std::pow( node.P_.Km_SERCA_, 2 ) + std::pow( calc, 2 ) );
+  const double m_inf = ip3 / ( ip3 + node.P_.Kd_IP3_1_ );
+  const double n_inf = calc / ( calc + node.P_.Kd_act_ );
+  const double calc_ER = ( node.P_.Ca_tot_ - calc ) / node.P_.ratio_ER_cyt_;
+  const double J_leak = node.P_.ratio_ER_cyt_ * node.P_.rate_L_ * ( calc_ER - calc );
+  const double J_channel = node.P_.ratio_ER_cyt_ * node.P_.rate_IP3R_ * std::pow( m_inf, 3 ) * std::pow( n_inf, 3 )
+    * std::pow( h_ip3r, 3 ) * ( calc_ER - calc );
+
+  f[ S::IP3 ] = ( node.P_.IP3_0_ - ip3 ) / node.P_.tau_IP3_;
+  f[ S::Ca ] = J_channel - J_pump + J_leak + node.B_.I_stim_;
+  f[ S::h_IP3R ] = alpha_h_ip3r * ( 1.0 - h_ip3r ) - beta_h_ip3r * h_ip3r;
+
+  return GSL_SUCCESS;
+}
+}
+
+/* ----------------------------------------------------------------
+ * Default constructors defining default parameters and state
+ * ---------------------------------------------------------------- */
+
+nest::astrocyte_lr_1994::Parameters_::Parameters_()
+  : Ca_tot_( 2.0 )      // uM
+  , IP3_0_( 0.16 )      // uM
+  , Kd_IP3_1_( 0.13 )   // uM
+  , Kd_IP3_2_( 0.9434 ) // uM
+  , Kd_act_( 0.08234 )  // uM
+  , Kd_inh_( 1.049 )    // uM
+  , Km_SERCA_( 0.1 )    // uM
+  , SIC_scale_( 1.0 )
+  , SIC_th_( 0.19669 )    // uM
+  , delta_IP3_( 5.0 )     // uM
+  , k_IP3R_( 0.0002 )     // 1/(uM*ms)
+  , rate_IP3R_( 0.006 )   // 1/ms
+  , rate_L_( 0.00011 )    // 1/ms
+  , rate_SERCA_( 0.0009 ) // uM/ms
+  , ratio_ER_cyt_( 0.185 )
+  , tau_IP3_( 7142.0 ) // ms
+{
+}
+
+nest::astrocyte_lr_1994::State_::State_( const Parameters_& p )
+{
+  y_[ IP3 ] = p.IP3_0_;
+  y_[ Ca ] = 0.073;
+  y_[ h_IP3R ] = 0.793;
+  y_[ SIC ] = 0.0;
+  y_[ DSIC ] = 0.0;
```

Comment:
Remove these as they are not being used in the dynamics function.

Reply:
Yes they are removed now.